client/src/lib/interface.ts

```diff
@@ -106,3 +129,146 @@ export interface Pin {
 }
 
 export type PinStatus = 'queued' | 'pinning' | 'pinned' | 'failed'
+
+/**
+ * This is an input used to construct the Token metadata as per EIP-1155
+ * @see https://eips.ethereum.org/EIPS/eip-1155#metadata
+ */
+export interface TokenInput {
+  /**
+   * Identifies the asset to which this token represents
+   */
+  name: string
+  /**
+   * Describes the asset to which this token represents
+   */
+  description: string
+  /**
+   * An `File` with mime type image/* representing the asset this
+   * token represents. Consider making any images at a width between `320` and
+   * `1080` pixels and aspect ratio between `1.91:1` and `4:5` inclusive.
+   *
+   * If `File` object is used, URL in the metadata will include a filename
+   * e.g. `ipfs://bafy...hash/cat.png`. If `Blob` is used URL in the metadata
+   * will not include filename or extension e.g. `ipfs://bafy...img/`
+   */
+  image: Blob | File
+
+  /**
+   * The number of decimal places that the token amount should display - e.g.
+   * `18`, means to divide the token amount by `1000000000000000000` to get its
+   * user representation.
+   */
+  decimals?: number
+
+  /**
+   * Arbitrary properties. Values may be strings, numbers, nested object or
+   * arrays of values. It is possible to provide a `File` or a `Blob` instance
+   * as property value, in which case it is stored on IPFS and metadata will
+   * contain URL to it in form of `ipfs://bafy...hash/name.png` or
+   * `ipfs://bafy...file/` respectively.
+   */
+  properties?: Object
+
+  localization?: Localization
+}
+
+interface Localization {
+  /**
+   * The URI pattern to fetch localized data from. This URI should contain the
+   * substring `{locale}` which will be replaced with the appropriate locale
+   * value before sending the request.
+   */
+  uri: string
+  /**
+   * The locale of the default data within the base JSON
+   */
+  default: string
+  /**
+   * The list of locales for which data is available. These locales should
+   * conform to those defined in the Unicode Common Locale Data Repository
+   * (http://cldr.unicode.org/).
+   */
+  locales: string[]
+}
+
+export interface StoreResult<T extends TokenInput> {
+  /**
+   * CID for the token that encloses all of the files including metadata.json
+   * for the stored token.
+   */
+  ipld: CID
```

Comment:
IDK, for discussion...I think people are less likely to know what IPLD is.

```suggestion
  ipfs: CID
```

Reply:
I honestly think it should be `cid`, but @mikeal thought `ipld` was better. If you both agree on `ipfs` that's good enough for me.